test/bundler-integration/smart-account/SA.concurrent.txn.specs.ts

```diff
@@ -0,0 +1,175 @@
+  let environment: BundlerTestEnvironment;
+
+  before(async function () {
+    const chainId = (await ethers.provider.getNetwork()).chainId;
+    if (chainId !== BundlerTestEnvironment.BUNDLER_ENVIRONMENT_CHAIN_ID) {
+      this.skip();
+    }
+
+    environment = await BundlerTestEnvironment.getDefaultInstance();
+  });
+
+  beforeEach(async function () {
+    [deployer, smartAccountOwner, charlie, verifiedSigner] =
+      await ethers.getSigners();
+  });
+
+  afterEach(async function () {
+    const chainId = (await ethers.provider.getNetwork()).chainId;
+    if (chainId !== BundlerTestEnvironment.BUNDLER_ENVIRONMENT_CHAIN_ID) {
+      this.skip();
+    }
+
+    await Promise.all([
+      environment.revert(environment.defaultSnapshot!),
+      environment.resetBundler(),
+    ]);
+  });
+
+  const setupTests = deployments.createFixture(
+    async ({ deployments, getNamedAccounts }) => {
+      await deployments.fixture();
+
+      const mockToken = await getMockToken();
+
+      const ecdsaModule = await getEcdsaOwnershipRegistryModule();
+      const EcdsaOwnershipRegistryModule = await ethers.getContractFactory(
+        "EcdsaOwnershipRegistryModule"
+      );
+
+      const ecdsaOwnershipSetupData =
+        EcdsaOwnershipRegistryModule.interface.encodeFunctionData(
+          "initForSmartAccount",
+          [await smartAccountOwner.getAddress()]
+        );
+      const smartAccountDeploymentIndex = 0;
+      const userSA = await getSmartAccountWithModule(
+        ecdsaModule.address,
+        ecdsaOwnershipSetupData,
+        smartAccountDeploymentIndex
+      );
+
+      await deployer.sendTransaction({
+        to: userSA.address,
+        value: ethers.utils.parseEther("10"),
+      });
+
+      await mockToken.mint(userSA.address, ethers.utils.parseEther("1000000"));
+
+      return {
+        entryPoint: await getEntryPoint(),
+        smartAccountImplementation: await getSmartAccountImplementation(),
+        smartAccountFactory: await getSmartAccountFactory(),
+        mockToken: mockToken,
+        ecdsaModule: ecdsaModule,
+        userSA: userSA,
+        verifyingPaymaster: await getVerifyingPaymaster(
+          deployer,
+          verifiedSigner
+        ),
+      };
+    }
+  );
+
+  it("Can send an ERC20 Transfer and native transfer userOps using 2D nonce", async () => {
+    const { entryPoint, mockToken, userSA, ecdsaModule } = await setupTests();
+
+    const charlieTokenBalanceBefore = await mockToken.balanceOf(
+      charlie.address
+    );
+    const tokenAmountToTransfer = ethers.utils.parseEther("0.5345");
+
+    const userOp1 = await makeEcdsaModuleUserOp(
+      "execute_ncC",
+      [
+        mockToken.address,
+        ethers.utils.parseEther("0"),
+        encodeTransfer(charlie.address, tokenAmountToTransfer.toString()),
+      ],
+      userSA.address,
+      smartAccountOwner,
+      entryPoint,
+      ecdsaModule.address,
+      {
+        preVerificationGas: 50000,
+      },
+      5 // nonceKey
```

Comment:
I would suggest making 
```
const erc20TransferUserOpNonceKey = 5;
```
instead of adding comment. and same for the second userOp

Reply:
done